app/package.json

```diff
@@ -34,23 +34,24 @@
     "@near-wallet-selector/sender": "^8.5.1",
     "@near-wallet-selector/wallet-connect": "^8.5.1",
     "@popperjs/core": "^2.11.5",
-    "@web3modal/wagmi": "^3.3.2",
     "@radix-ui/react-accordion": "^1.1.2",
     "@radix-ui/react-dialog": "^1.0.5",
     "@radix-ui/react-dropdown-menu": "^2.0.6",
     "@radix-ui/react-icons": "^1.3.0",
     "@supabase/supabase-js": "^2.33.1",
     "@types/js-cookie": "^3.0.3",
+    "@web3modal/wagmi": "^3.3.2",
     "ag-grid-community": "^27.3.0",
     "ag-grid-react": "^27.3.0",
     "axios": "^1.5.0",
     "bn.js": "^5.2.0",
     "chart.js": "^3.8.2",
     "chartjs-adapter-moment": "^1.0.0",
-    "clipboard": "^2.0.11",
     "class-variance-authority": "^0.7.0",
+    "clipboard": "^2.0.11",
     "clsx": "^2.0.0",
     "dropzone": "^6.0.0-beta.2",
+    "ethers": "^6.9.0",
```

Comment:
Although using ethers is a good approach but i was thinking of using viem (https://viem.sh/docs/introduction.html)

- viem can infer types based on ABI 
- very less bundle size than ethers
- created and recommended by 'wagmi' package authors which we are using in our app.

Reply:
@netpoe and we are not utilising wagmi properly. we can create our custom hooks (using wagmi hooks) instead of class interface.Hooks would be much faster. calling a contract method is also very simple https://www.0xdev.co/how-to-interact-with-smart-contracts-using-wagmi-in-react/ and handle cases like error, loading and return types